### PR TITLE
azfile: Share Client

### PR DESCRIPTION
**Pull request by souravgupta-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azfile/internal/testcommon/clients_auth.go [hunk 1]
```diff
@@ -39,6 +42,15 @@ const (
 	FakeStorageURL     = "https://fakestorage.file.core.windows.net"
 )
 
+var (
+	SampleSDDL = `O:S-1-5-32-548G:S-1-5-21-397955417-626881126-188441444-512D:(A;;RPWPCCDCLCSWRCWDWOGA;;;S-1-0-0)`
```
Comment (vibhansa-msft) [inline sdk/storage/azfile/internal/testcommon/clients_auth.go:46]: This is not used anywhere?

File: sdk/storage/azfile/sas/query_params.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 // timeFormat represents the format of a SAS start or expiry time. Use it when formatting/parsing a time.Time.
 const (
 	timeFormat         = "2006-01-02T15:04:05Z" // "2017-07-27T00:00:00Z" // ISO 8601
-	snapshotTimeFormat = "2006-01-02T15:04:05.0000000Z07:00"
+	SnapshotTimeFormat = "2006-01-02T15:04:05.0000000Z07:00"
```
Comment (vibhansa-msft) [inline sdk/storage/azfile/sas/query_params.go:20]: Is this variable exposed from this package ?

File: sdk/storage/azfile/internal/testcommon/clients_auth.go (hunk 1 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/internal/testcommon/clients_auth.go:46]: This is used in `TestSharePutAndGetPermission` in share client.

File: sdk/storage/azfile/sas/query_params.go (hunk 2 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/query_params.go:20]: This variable is used by the share client's `GetSASURL()` method to parse the share snapshot time to its appropriate format.

File: sdk/storage/azfile/service/client_test.go [hunk 3]
```diff
@@ -253,29 +251,78 @@ func (s *ServiceUnrecordedTestsSuite) TestSASServiceClient() {
...
 	resp, err := svcClient.GetProperties(context.Background(), nil)
 	_require.NoError(err)
 	_require.NotNil(resp.RequestID)
+
+	// create share using account SAS
+	_, err = svcClient.CreateShare(context.Background(), shareName, nil)
+	_require.NoError(err)
+
+	defer func() {
+		_, err := svcClient.DeleteShare(context.Background(), shareName, nil)
```
Comment (gapra-msft) [inline sdk/storage/azfile/service/client_test.go:268]: won't this defer Delete Share come back with an error since you are Deleting below?

Comment (gapra-msft) [reply sdk/storage/azfile/service/client_test.go:268]: oh  nvm I see the restore share

File: sdk/storage/azfile/share/client_test.go [hunk 4]
```diff
@@ -0,0 +1,1343 @@
...
+	_require.NoError(err)
+	_require.Equal(*getStats.ShareUsageBytes, int64(1024*1024*1024*1024))
+}*/
+
+func deleteShare(ctx context.Context, _require *require.Assertions, shareClient *share.Client, o *share.DeleteOptions) {
+	_, err := shareClient.Delete(ctx, o)
+	_require.NoError(err)
+}
+
+func (s *ShareUnrecordedTestsSuite) TestShareCreateSnapshotNonDefault() {
```
Comment (gapra-msft) [inline sdk/storage/azfile/share/client_test.go:930]: Should some of these tests be recorded? Or is the idea to record all at once in the end?

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azfile/share/client_test.go (hunk 4 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/share/client_test.go:930]: Yes, will record the tests in some future PR.

File: sdk/storage/azfile/sas/service.go [hunk 5]
```diff
@@ -38,8 +39,8 @@ type FileSignatureValues struct {
 
 // SignWithSharedKey uses an account's SharedKeyCredential to sign this signature values to produce the proper SAS query parameters.
 func (v FileSignatureValues) SignWithSharedKey(sharedKeyCredential *SharedKeyCredential) (QueryParameters, error) {
```
Comment (jhendrixMSFT) [inline sdk/storage/azfile/sas/service.go:41]: It seems a bit odd to call this `FileSignatureValues` when it supports files and shares.  What about simply `SignatureValues`?

Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/service.go:41]: Its like how we have in azblob where `BlobSignatureValues` supports both container and blobs.
I can rename it to `SignatureValues` if that's what we want in azfile.

Comment (souravgupta-msft) [reply sdk/storage/azfile/sas/service.go]: Done

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**